modules/tinymce/src/core/test/ts/browser/content/InsertContentTest.ts

```diff
@@ -1,214 +1,329 @@
-import { Pipeline } from '@ephox/agar';
-import { UnitTest } from '@ephox/bedrock-client';
-import { LegacyUnit, TinyLoader } from '@ephox/mcagar';
+import { Log, Pipeline, Step } from '@ephox/agar';
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import { TinyApis, TinyLoader } from '@ephox/mcagar';
 import Editor from 'tinymce/core/api/Editor';
 import * as InsertContent from 'tinymce/core/content/InsertContent';
 import Theme from 'tinymce/themes/silver/Theme';
 
 UnitTest.asynctest('browser.tinymce.core.content.InsertContentTest', (success, failure) => {
-  const suite = LegacyUnit.createSuite<Editor>();
-
   Theme();
 
-  const assertSelection = function (editor, selector, offset) {
-    const node = editor.$(selector)[0];
-    const rng = editor.selection.getRng();
-
-    LegacyUnit.equalDom(rng.startContainer, node.firstChild);
-    LegacyUnit.equal(rng.startOffset, offset);
-    LegacyUnit.equal(rng.collapsed, true);
-  };
-
-  suite.test('insertAtCaret - i inside text, converts to em', function (editor) {
-    editor.setContent('<p>1234</p>');
-    editor.focus();
-    LegacyUnit.setSelection(editor, 'p', 2);
-    InsertContent.insertAtCaret(editor, '<i>a</i>');
-    LegacyUnit.equal(editor.getContent(), '<p>12<em>a</em>34</p>');
+  const sInsert = (editor: Editor, value: string | { content: string; paste?: boolean; merge?: boolean }) => Step.sync(() => {
+    InsertContent.insertAtCaret(editor, value);
   });
 
-  suite.test('insertAtCaret - ul at beginning of li', function (editor) {
-    editor.setContent('<ul><li>12</li></ul>');
-    editor.focus();
-    LegacyUnit.setSelection(editor, 'li', 0);
-    InsertContent.insertAtCaret(editor, { content: '<ul><li>a</li></ul>', paste: true });
-    LegacyUnit.equal(editor.getContent(), '<ul><li>a</li><li>12</li></ul>');
-    assertSelection(editor, 'li:nth-child(2)', 0);
-  });
+  const getTests = (editor: Editor, tinyApis: TinyApis) => [
+    Log.stepsAsStep('TBA', 'insertAtCaret - i inside text, converts to em', [
+      tinyApis.sSetContent('<p>1234</p>'),
+      tinyApis.sFocus(),
+      tinyApis.sSetSelection([ 0, 0 ], 2, [ 0, 0 ], 2),
```

Comment:
Should we make a selector-based alternative to this in a future PR?

Reply:
We should, but that's way outside the scope of this PR. Also remember that a selector based lookup doesn't really work for text nodes, which is what we want the selection to be in, in most cases. So we'd need to discuss how we'd do that.